platforms/nuttx/CMakeLists.txt

```diff
@@ -241,8 +257,71 @@ if (NOT CONFIG_BUILD_FLAT)
 
 		DEPENDS px4 px4_kernel
 	)
-
 else()
+## TODO REMOVE ME
```

Comment:
?

Reply:
This PR needs clean up, I posted it just to a) keep the changes somewhere safe and b) to get initial comments about the changes. Obviously I will address all of the TODOs and quirks.